src/core/init/manifest_update_scheduler.ts

```diff
@@ -93,20 +104,37 @@ export default function manifestUpdateScheduler({
   // The Manifest always keeps the same Manifest
   const { manifest } = initialManifest;
 
+  /** Number of consecutive times the parsing has been done in `unsafeMode`. */
+  let consecutiveUnsafeMode = 0;
   function handleManifestRefresh$(
     manifestInfos: { manifest: Manifest;
                      sendingTime?: number;
                      receivedTime? : number;
                      parsingTime : number;
                      updatingTime? : number; }): Observable<IWarningEvent> {
-    const { sendingTime } = manifestInfos;
+    const { sendingTime,
+            parsingTime,
+            updatingTime } = manifestInfos;
+
+    /**
+     * Total time taken to fully update the last Manifest.
+     * Note: this time also includes possible requests done by the parsers.
+     */
+    const totalUpdateTime = parsingTime + (updatingTime ?? 0);
+
+    // Only perform parsing in `unsafeMode` when the last full parsing took a
```

Comment:
/** */ ?

Reply:
/** */ would be for JSDoc, so it would be the actual documentation of the `unsafeModeEnabled` constant.

Here I felt it was more of a code comment explaining what I do here.